src/python/pants/backend/native/subsystems/native_build_settings.py

```diff
@@ -7,31 +7,26 @@
 from pants.subsystem.subsystem import Subsystem
 
 
-class NativeCompileSettings(Subsystem):
-  """Any settings relevant to a compiler invocation."""
+class NativeBuildSettings(Subsystem):
+  """Any settings relevant to a compiler and/or linker invocation."""
+  options_scope = 'native-build-settings'
```

Comment:
I think people don't think of the two languages as interchangeable because it has literally not been possible to formulate code as such without sacrificing a goat or two, even in the case of mixed C and C++ code. The reason for the merged subsystem was to make it clear that C and C++ are going to depend on each other the exact same way -- my goal is 0% to let it become a kitchen sink. I think `--fatal-warnings` is an example of something that could be split into C and C++-specific subsystems. Does that address your concern in:
> otherwise, doing a bit of work to identify which language a target is building and apply slightly different settings could be valuable in the long run (it's hard to split them later).

I think that `--strict-deps` as a concept doesn't really exist in traditional C/C++ build systems, and separately, I think if we can support zero-(user-)overhead interdependence of C and C++ targets, it would be nice to uphold. I can make it clear in the docstring that `NativeBuildSettings` is for "pants stuff and not things like compiler arguments".

> Anecdotally: Java and Scala having different strict_deps settings has been a lifesaver, because Scala's compiler has drastically different dep requirements, and it's a much larger step to enable strict deps there.
Could you elaborate on this? It is currently my impression (as in the above paragraph) that because `--strict-deps` is already this new thing we are introducing into C/C++, and is entirely managed by pants, it makes *more* sense to merge that setting. What does "drastically different dep requirements" mean?

Reply:
Also, I wouldn't really call this "modelled" after the Java and Scala subsystems -- I had already made up a quick subsystem base class to calculate subsystem/target mirrored options, I added a bool option to it, and used the existing strict deps calculation methods. I had originally done this before you did the zinc upgrade and that merge conflict was *fun*.

Reply:
I want you to know that it took all of my strength to not implement `--compiler-option-sets` just now but I saved that for later because this PR otherwise works.

Separately, the test failure (in testprojects integration) is because we don't currently have a way to allow any command-line option to supersede a target option like `--strict-deps` (which was the idea, but it makes it hard to test). I will figure out a low-overhead solution that allows us to have these building in that integration test.

Reply:
I think I'll just rewrite that BUILD file temporarily in the test, that's easy.